components/webui/client/src/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {useEffect} from "react";
+
+import {
+    Editor,
+    EditorProps,
+    useMonaco,
+} from "@monaco-editor/react";
+import {language as sqlLanguage} from "monaco-editor/esm/vs/basic-languages/sql/sql.js";
+
+import "./monaco-config";
+
+
+type SqlEditorProps = Omit<EditorProps, "language">;
+
+/**
+ * Monaco editor with highlighting and autocomplete for SQL syntax.
+ *
+ * @param props
+ * @return
+ */
+const SqlEditor = (props: SqlEditorProps) => {
+    const monaco = useMonaco();
+
+
+    useEffect(() => {
+        if (null === monaco) {
+            return () => {
+            };
+        }
+
+        // Adds autocomplete suggestions for SQL keywords on editor load
+        const provider = monaco.languages.registerCompletionItemProvider("sql", {
```

Comment:
for lazy users who don't even wanna type a single character
```suggestion
        const provider = monaco.languages.registerCompletionItemProvider("sql", {
            triggerCharacters: [
                " ",
                "\n",
            ],
```